### DISCUSSION TITLE
Are the libc++ libraries in different ndk versions compatible with each other?

### DISCUSSION THREAD

**Discussion by liuxiaoli007** [android/ndk#1694] [Q&A]
Are the libc++ libraries in different ndk versions compatible with each other? 
For example，My apk uses libc++_shared.so, and I have integrated a three-party ndk, which also uses the libc++ dynamic library, which ndk version should I choose to build libc++_shared.so?

**Comment by DanAlbert** [comment]
Typically yes. There was a major break between r10 and r11 for example, but we avoid them wherever possible. This isn't a thoroughly tested configuration (we have a lot of tests, but they're all testing the same version of the NDK) so it's definitely possible for this to happen by accident, but we would consider it a major bug if it were reported.

If your third-party code is being integrated with a [prefab AAR](https://developer.android.com/studio/build/dependencies#publish-native-libs-in-aars), prefab will keep track of this for you. If there's ever a compatibility break that you need to worry about, prefab will check for that (assuming you're using a version of prefab that's new enough to know about the break, but as long as you're staying up to date on AGP you'll also be up to date on prefab).

**Comment by promanowicz** [comment]
@liuxiaoli007  You might take a look at this issue, it might be interesting for you.
https://github.com/realm/realm-kotlin/issues/694
